metadata-ingestion/src/datahub/utilities/sql_lineage_parser_impl.py

```diff
@@ -32,18 +35,59 @@ def __init__(self, sql_query: str) -> None:
         if "lateral flatten" in sql_query:
             sql_query = sql_query[: sql_query.find("lateral flatten")]
 
+        # BigQuery can use # as a comment symbol and that may break the parsing
+        # if the comment starts without a space sign after #
+        # and the comment is written inside SQL DDL statement (e.g., after CREATE...AS)
+        sql_query = re.sub(r"#([^ ])", r"# \1", sql_query, flags=re.IGNORECASE)
+
+        # Wrap calls to field '<table_alias>.from' in ` so it would not be taken as a reserved keyword
+        sql_query = re.sub(r"(\w*\.from)", r"`\1`", sql_query, flags=re.IGNORECASE)
+
+        # Apply sqlparser formatting to get rid of comments and reindent keywords
+        # which should remove some potential inconsistencies in parsing output
+        sql_query = sqlparse.format(
+            sql_query.strip(),
+            reindent_aligned=True,
+            strip_comments=True,
+        )
+
+        # SqlLineageParser does not allow table/view names not being wrapped in quotes or backticks
+        # Add ` signs before and after supposed object name that comes right after reserved word FROM
```

Comment:
Will these show up in the output? Will backticks make sense for all SQL sources or just BigQuery?

Reply:
It's a standard for BigQuery, although sqllineage app at https://sqllineage.herokuapp.com/ was also allowing them as object identifiers, although some other SQL standards would use single quotes, for instance.

Can't say if backticks are friendly in other sources.

Reply:
@anshbansal what do you think about the following:
We can move the changes of this PR from sql_lineage_parser_impl towards a new wrapper class like BigQuerySqlParser inside sql_parser. That new class can use be almost like DefaultSqlParser but with a set of regex-based functions to make additional BigQuery-related sql parses before sending the query to SqlLineageSQLParser and the multiprocessing queue.

Something like:
```
class BigQuerySQLParser(SQLParser):
    parser: SQLParser

    def __init__(self, sql_query: str) -> None:
        super().__init__(sql_query)
        
        self.parsed_sql_query = NewRegexBasedParseFunction(sql_query)
        self.parser = SqlLineageSQLParser(self.parsed_sql_query)

    def get_tables(self) -> List[str]:
        return self.parser.get_tables()

    def get_columns(self) -> List[str]:
        return self.parser.get_columns()

```
That would keep SqlLineageSQLParser friendly to other sources while BigQuery still gets its special treatment.

Reply:
Sounds good to me. 
Please keep that inside a separate file not `sql_parser.py` in case we later decide to start importing bigquery specific libraries here.

Reply:
Done, new commit is here